package.json

```diff
@@ -16,19 +16,27 @@
         "node": ">=14 <17"
     },
     "scripts": {
-        "copy-scripts": "yarn copy-scripts:array && yarn copy-scripts:recorder",
+        "copy-scripts": "mkdir -p frontend/dist/ && yarn copy-scripts:array && yarn copy-scripts:recorder",
         "copy-scripts:array": "cp node_modules/posthog-js/dist/array.js* frontend/dist/",
         "copy-scripts:recorder": "cp node_modules/rrweb/dist/rrweb.min.js frontend/dist/recorder.js && cp node_modules/rrweb/dist/rrweb.min.js.map frontend/dist/recorder.js.map && sed -i -e s/rrweb.min.js.map/recorder.js.map/ frontend/dist/recorder.js",
         "test": "jest",
-        "start": "concurrently -n WEBPACK,TYPEGEN -c blue,green \"yarn run start-http\" \"yarn run typegen:watch\"",
-        "start-http": "mkdir -p frontend/dist/ && cp -a frontend/public/* frontend/dist/ && npm run copy-scripts && webpack serve",
-        "start-https": "mkdir -p frontend/dist/ && cp -a frontend/public/* frontend/dist/ && npm run copy-scripts && webpack serve --https",
-        "start-docker": "mkdir -p frontend/dist/ && cp -a frontend/public/* frontend/dist/ && npm run copy-scripts && webpack serve --host 0.0.0.0",
+        "start": "concurrently -n ESBUILD,TYPEGEN -c yellow,green \"yarn start-http\" \"yarn run typegen:watch\"",
+        "start-http": "yarn start-http:esbuild",
+        "start-http:esbuild": "yarn clean && yarn copy-scripts && node frontend/build.mjs --dev",
+        "start-http:webpack": "mkdir -p frontend/dist/ && cp -a frontend/public/* frontend/dist/ && npm run copy-scripts && webpack serve",
+        "start-https": "yarn start-https:webpack",
+        "start-https:webpack": "yarn start-http:webpack --https",
+        "start-docker": "yarn start-docker:esbuild",
+        "start-docker:esbuild": "yarn start-http:esbuild --host 0.0.0.0",
+        "start-docker:webpack": "yarn start-http:webpack --host 0.0.0.0",
```

Comment:
I tried not to be too disruptive here, and duplicated scripts wherever possible. With very quick changes in the scripts below we can switch the default build engine back to webpack.